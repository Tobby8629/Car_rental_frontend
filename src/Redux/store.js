import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import CarSlice from './CarSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Cars: CarSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
