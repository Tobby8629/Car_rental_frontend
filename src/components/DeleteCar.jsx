import React from 'react';

function DeleteCar() {
  return (
    <section>DeleteCar</section>
  );
}

export default DeleteCar;
