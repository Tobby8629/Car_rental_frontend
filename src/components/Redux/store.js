import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
