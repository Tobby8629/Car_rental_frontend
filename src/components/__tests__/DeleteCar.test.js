import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../Redux/store';
import DeleteCar from '../DeleteCar';

test('home component match snapshot', () => {
  const user = { id: '123' }; // Set the desired user ID here
  const localStorageMock = {
    getItem: jest.fn().mockReturnValue(JSON.stringify(user)),
  };
  Object.defineProperty(window, 'localStorage', {
    value: localStorageMock,
    writable: true,
  });

  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteCar />
      </BrowserRouter>
    </Provider>,

  );
  expect(asFragment()).toMatchSnapshot();
});
