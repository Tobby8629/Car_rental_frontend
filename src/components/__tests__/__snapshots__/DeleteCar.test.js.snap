// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component match snapshot 1`] = `
<DocumentFragment>
  <section
    class="delete myreserve"
  >
    <div>
      <h2>
        You do not have any car, please add a
        <a
          href="/add"
        >
           new car
        </a>
      </h2>
    </div>
  </section>
</DocumentFragment>
`;
